internal/services/notification/webhooks.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2023 Sorint.lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package notification
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/sorintlab/errors"
+
+	"agola.io/agola/internal/services/common"
+	"agola.io/agola/internal/services/gateway/action"
+	rstypes "agola.io/agola/services/runservice/types"
+)
+
+const (
+	signatureSHA256Key = "X-Agola-SHA256Signature"
+
+	agolaEventHeader = "X-Agola-Event"
+
+	webhookVersion = 1
+)
+
+type RunWebhook struct {
+	// version of the webhook struct data
+	Version uint64 `json:"version"`
+
+	// info of the project
+	ProjectInfo ProjectInfo `json:"project_info"`
+
+	// current run status
+	Run *Run `json:"run"`
+
+	// the trigger of the event
+	// (i.e. the value run_phase_changed indicate a run phase changed event)
+	EventType string `json:"event_type"`
```

Comment:
Use a dedicated enum for EventType/Action and add the description there and not here